libnfs4acl/nfs4_acl_utils.c

```diff
@@ -58,33 +58,40 @@ inline struct nfs4_ace* nfs4_get_next_ace(struct nfs4_ace **ace)
 
 struct nfs4_ace* nfs4_get_ace_at(struct nfs4_acl *acl, unsigned int index)
 {
-	struct nfs4_ace *ace;
+	struct nfs4_ace *ace = NULL;
 	int i;
 
-	if (index >= acl->naces)
+	if (index >= acl->naces) {
+		errno = E2BIG;
 		return NULL;
+	}
 
 	ace = nfs4_get_first_ace(acl);
-	for (i = 0; i < index; i++)
+	for (i = 0; i < index; i++) {
 		ace = nfs4_get_next_ace(&ace);
+	}
 
 	return ace;
 }
 
 int nfs4_insert_ace_at(struct nfs4_acl *acl, struct nfs4_ace *ace, unsigned int index)
 {
-	struct nfs4_ace *ap;
+	struct nfs4_ace *ap = NULL;
 
-	if (acl == NULL || ace == NULL || index > acl->naces)
+	if (acl == NULL || ace == NULL || index > acl->naces) {
+		errno = E2BIG;
+		printf("insert ace at acl; %p, ace: %p, index: [%d], naces: [%d] \n", acl, ace, index, acl ? acl->naces : 0);
```

Comment:
I imagine this is another rare scenario but curious if this is leftover debug or left in to alert end-user when something is wrong.

Reply:
It can possibly be reached if has manually specified an ACL index that is too large. Possibly reachable by setfacl. Will think about a better log message.